blueprints/data-solutions/vertex-mlops/vertex.tf

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_vertex_ai_metadata_store" "store" {
+  provider    = google-beta
+  project     = module.project.project_id
+  name        = "default"
+  description = "Vertex Ai Metadata Store"
+  region      = var.region
+  dynamic "encryption_spec" {
+    for_each = try(var.service_encryption_keys.aiplatform, null) == null ? [] : [""]
+
+    content {
+      kms_key_name = try(var.service_encryption_keys.aiplatform, null)
+    }
+  }
+  # `state` value will be decided automatically based on the result of the configuration
+  lifecycle {
+    ignore_changes = [state]
+  }
+}
+
+module "service-account-notebook" {
+  source     = "../../../modules/iam-service-account"
+  project_id = module.project.project_id
+  name       = "notebook-sa"
+}
+
+resource "google_notebooks_runtime" "runtime" {
+  for_each = { for k, v in var.notebooks : k => v if v.type == "MANAGED" }
+  name     = "${var.prefix}-${each.key}"
+
+  project  = module.project.project_id
+  location = var.region
+  access_config {
+    access_type   = "SINGLE_USER"
+    runtime_owner = try(var.notebooks[each.key].owner, null)
+  }
+  software_config {
+    enable_health_monitoring = true
+  }
+  virtual_machine {
+    virtual_machine_config {
+      machine_type     = var.notebooks[each.key].machine_type
+      network          = local.vpc
+      subnet           = local.subnet
+      internal_ip_only = var.notebooks[each.key].internal_ip_only
+      dynamic "encryption_config" {
+        for_each = try(local.service_encryption_keys.notebooks, null) == null ? [] : [1]
+        content {
+          kms_key = local.service_encryption_keys.notebooks
+        }
+      }
+      metadata = {
+        notebook-disable-nbconvert = "false"
+        notebook-disable-downloads = "true"
+        notebook-disable-terminal  = "false"
+        notebook-disable-root      = "true"
+      }
+      data_disk {
+        initialize_params {
+          disk_size_gb = "100"
+          disk_type    = "PD_STANDARD"
+        }
+      }
+    }
+  }
+}
```

Comment:
all the hardcoded values in this resource should probably become their own variable

Reply:
I will address this is a future PR.